bool flying
bool hovering 
bool position
bool local